.github/workflows/build.yml

```diff
@@ -0,0 +1,144 @@
+      matrix:
+        php-versions: ['7.3', '7.4']
+        dependencies: ['latest']
+        composer-arguments: [''] # to run --ignore-platform-reqs in experimental builds
+        static-analysis: ['no']
+        experimental: [false]
+        include:
+          - php-versions: '7.3'
+            static-analysis: 'psalm'
+            experimental: false
+            dependencies: 'latest'
+
+          # Experimental build for PHP 8
+          #- php-versions: '8.0'
+          #  composer-arguments: '--ignore-platform-reqs'
+          #  static-analysis: 'no'
+          #  experimental: true
+          #  dependencies: 'latest'
+
+          # Build for minimum dependencies. Fails right now, hence deactivated.
+          #- php-versions: '7.3'
+          #  static-analysis: 'no'
+          #  experimental: false
+          #  dependencies: 'lowest'
+
+    runs-on: ubuntu-latest
+    services:
+      mariadb:
+        image: mariadb:10.2
+        env:
+          MYSQL_USER: neos
+          MYSQL_PASSWORD: neos
+          MYSQL_DATABASE: flow_functional_testing
+          MYSQL_ROOT_PASSWORD: neos
+        ports:
+          - "3306:3306"
+        options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3
+
+    env:
+      FLOW_TARGET_VERSION: ${{ github.base_ref }}
+      FLOW_TARGET_REPOSITORY: ${{ github.repository }}
+      FLOW_CONTEXT: Testing
+      FLOW_DIST_FOLDER: flow-development-distribution
+      FLOW_FOLDER: flow-development-collection
+
+    defaults:
+      run:
+        working-directory: ${{ env.FLOW_DIST_FOLDER }}
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        path: ${{ env.FLOW_FOLDER }}
+
+    - name: Setup PHP
+      uses: shivammathur/setup-php@v2
+      with:
+        php-version: ${{ matrix.php-versions }}
+        extensions: mbstring, xml, json, zlib, iconv, intl, pdo_sqlite, mysql, redis, memcached
+        coverage: xdebug #optional
+        ini-values: opcache.fast_shutdown=0
+
+    - name: Checkout development distribution
+      uses: actions/checkout@v2
+      with:
+        repository: neos/flow-development-distribution
+        ref: ${{ env.FLOW_TARGET_VERSION }}
+        path: ${{ env.FLOW_DIST_FOLDER }}
+
+    - name: Set alias branch name
+      run: if [ "${FLOW_TARGET_VERSION}" == "master" ]; then echo "FLOW_BRANCH_ALIAS=dev-master"; else echo "FLOW_BRANCH_ALIAS=${FLOW_TARGET_VERSION}.x-dev"; fi >> $GITHUB_ENV
+
+    - name: Update composer.json
+      run: |
+        git -C ../${{ env.FLOW_FOLDER }} checkout -b build
+        composer config repositories.flow '{ "type": "path", "url": "../${{ env.FLOW_FOLDER }}", "options": { "symlink": false } }'
+        composer require --no-update ${{ github.repository }}:"dev-build as ${{ env.FLOW_BRANCH_ALIAS }}"
+
+    - name: Cache Composer packages
+      id: composer-cache
+      uses: actions/cache@v2
+      with:
+        path: ${{ env.FLOW_DIST_FOLDER }}
```

Comment:
NOTE: This caches the full distribution installation. This means, the dependencies will not be updated unless the composer.json files are changed.
The alternative is to use `path: ~/.composer/cache/files` and comment the `if:` in the install dependencies step.